engine/gui/src/dmsdk/gui/gui.h

```diff
@@ -0,0 +1,241 @@
+        ADJUST_REFERENCE_PARENT   = 1,
+        ADJUST_REFERENCE_DISABLED = 2
+    };
+
+    /*#
+     * This enum denotes what kind of texture type the m_Texture pointer is referencing.
+     * @name NodeTextureType
+     * @type enum
+     * @member NODE_TEXTURE_TYPE_NONE
+     * @member NODE_TEXTURE_TYPE_TEXTURE
+     * @member NODE_TEXTURE_TYPE_TEXTURE_SET
+     * @member NODE_TEXTURE_TYPE_DYNAMIC
+     */
+    enum NodeTextureType
+    {
+        NODE_TEXTURE_TYPE_NONE,
+        NODE_TEXTURE_TYPE_TEXTURE,
+        NODE_TEXTURE_TYPE_TEXTURE_SET,
+        NODE_TEXTURE_TYPE_DYNAMIC
+    };
+
+    // NOTE: These enum values are duplicated in scene desc in gamesys (gui_ddf.proto)
+    // Don't forget to change gui_ddf.proto if you change here
+
+    /*#
+     * @name Result
+     * @type enum
+     * @member NODE_TYPE_BOX //!< 0,
+     * @member NODE_TYPE_TEXT //!< 1,
+     * @member NODE_TYPE_PIE //!< 2,
+     * @member NODE_TYPE_TEMPLATE //!< 3,
+     * @member NODE_TYPE_PARTICLEFX //!< 5,
+     * @member NODE_TYPE_CUSTOM //!< 6,
+     * @member NODE_TYPE_COUNT //!< 7,
+     */
+    enum NodeType
+    {
+        NODE_TYPE_BOX  = 0,
+        NODE_TYPE_TEXT = 1,
+        NODE_TYPE_PIE  = 2,
+        NODE_TYPE_TEMPLATE = 3,
+        //NODE_TYPE_SPINE = 4, // Deprecated, can we remove it from the ddf?
+        NODE_TYPE_PARTICLEFX = 5,
+        NODE_TYPE_CUSTOM = 6,
+        NODE_TYPE_COUNT = 7,
+    };
+
+
+    /*#
+     * @name Result
+     * @type enum
+     * @member NODE_TEXTURE_TYPE_NONE
+     * @member RESULT_OK //!< 0
+     * @member RESULT_SYNTAX_ERROR //!< -1
+     * @member RESULT_SCRIPT_ERROR //!< -2
+     * @member RESULT_OUT_OF_RESOURCES //!< -4
+     * @member RESULT_RESOURCE_NOT_FOUND //!< -5
+     * @member RESULT_TEXTURE_ALREADY_EXISTS //!< -6
+     * @member RESULT_INVAL_ERROR //!< -7
+     * @member RESULT_INF_RECURSION //!< -8
+     * @member RESULT_DATA_ERROR //!< -9
+     * @member RESULT_WRONG_TYPE //!< -10
+     */
+    enum Result
+    {
+        RESULT_OK = 0,
+        RESULT_SYNTAX_ERROR = -1,
+        RESULT_SCRIPT_ERROR = -2,
+        RESULT_OUT_OF_RESOURCES = -4,
+        RESULT_RESOURCE_NOT_FOUND = -5,
+        RESULT_TEXTURE_ALREADY_EXISTS = -6,
+        RESULT_INVAL_ERROR = -7,
+        RESULT_INF_RECURSION = -8,
+        RESULT_DATA_ERROR = -9,
+        RESULT_WRONG_TYPE = -10,
+    };
+
+    /*#
+     * @name NewNode
+     * @param scene [type:HScene] the gui scene
+     * @param position [type:dmVMath::Point3] the position
+     * @param size [type:dmVMath::Vector3] the size
+     * @param node_type [type:dmGui::NodeType] the node type
+     * @param custom_type [type:uint32_t] If node_type == dmGui::NODE_TYPE_CUSTOM, then this is used to create a custom node data for the registered custom type
+     * @return node [type: dmGui::HNode] the created node
+     */
+    HNode NewNode(HScene scene, const dmVMath::Point3& position, const dmVMath::Vector3& size, NodeType node_type, uint32_t custom_type);
```

Comment:
Although it's part of the public sdk due to extension-spine, we also probably don't want to promote it too much, as we want to remove the support for gui (in favour of equivalent functionality in the game object world)